### DISCUSSION TITLE
Does namespace have a specific purpose?

### DISCUSSION THREAD

**Discussion by prestonvasquez** [tmc/langchaingo#991] [Q&A]
I'm noticing that most `vectorstore` instrumentations (with the exception of `qdrant`) use `Store` as a wrapper for a client and typically use namespace as a collection analogue. 

For MongoDB, it makes sense to have a `Store` object wrap a `mongo.Collection` since adding and searching vectors is collection-specific. In this case `AddDocuments()` and `SimilaritySearch()` become analogues for `Collection.InsertMany()` and `Collection.Aggregate()`. This also keeps our implementation inline with [PY](https://github.com/langchain-ai/langchain/blob/f4ffd692a3548681b4d05c0f255a9b83c07a2e0b/libs/partners/mongodb/langchain_mongodb/vectorstores.py#L206) and [TS](https://github.com/langchain-ai/langchainjs/blob/abdcd1fa3b74f066e4a7cbe3688a99f719f8f75b/libs/langchain-mongodb/src/storage.ts#L53). Notably, a collection / namespace isn't actually optional and makes the API awkward:

```go
store, _ := mongovector.New(client)
if _, err := store.AddDocuments(ctx, docs); err != nil {
	panic(err) // This always errors since a collection / namespace has not been defined.
}
```

**TL;DR**: Is the `Namespace` option generic enough to use as an index name, or does this field have a specific but undocumented use case? 

For exampe:

```go
docs, _ := store.SimilaritySearch(ctx, "text", 20, vectorstore.WithNameSpace("some_index_to_search"))
```

**Comment by prestonvasquez** [comment]
CC: @blink1073 @matthewdale @qingyang-hu

**Reply by jtazin** [reply]
hi @FluffyKebab! looks like you added the [namespace option](https://github.com/tmc/langchaingo/commit/5acba9111efecc250d9f85e8b9a6e710640361e1), do you have any insight into @prestonvasquez's question?

**Comment by prestonvasquez** [comment]
We've decided to use the namespace option as a way for users to switch indexes at the operation leve.
